crates/derive/src/types/block.rs

```diff
@@ -24,6 +24,52 @@ impl BlockInfo {
     pub fn new(hash: B256, number: u64, parent_hash: B256, timestamp: u64) -> Self {
         Self { hash, number, parent_hash, timestamp }
     }
+
+    /// Returns the block ID.
+    pub fn id(&self) -> BlockID {
+        BlockID { hash: self.hash, number: self.number }
+    }
+}
+
+impl core::fmt::Display for BlockInfo {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(
+            f,
+            "BlockInfo {{ hash: {}, number: {}, parent_hash: {}, timestamp: {} }}",
+            self.hash, self.number, self.parent_hash, self.timestamp
+        )
+    }
+}
+
+/// Block ID identifies a block by its hash and number
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub struct BlockID {
+    /// The block hash
+    pub hash: BlockHash,
+    /// The block number
+    pub number: BlockNumber,
+}
+
+impl core::fmt::Display for BlockID {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f, "{{ hash: {}, number: {} }}", self.hash, self.number)
+    }
+}
+
+/// An L2 Block Ref
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub struct L2BlockRef {
```

Comment:
Looks like some types got dup'd when doing parallel work. Can we replace the `L2BlockInfo` type w/ this or vice versa?

Reply:
Ah missed that, thanks! Bad rebase/merge